src/base/bittorrent/peerinfo.cpp

```diff
@@ -408,3 +408,8 @@ QString PeerInfo::flagsDescription() const
 {
     return m_flagsDescription;
 }
+
+int PeerInfo::downloadingPieceIndex() const
```

Comment:
Returning to our discussion about the right design...
Instead to create a logical for external code  `QStringList PeerInfo::downloadingFiles()` method, you put in the interface `int PeerInfo::downloadingPieceIndex()` method which nobody wants and which in this case can be attributed to the implementation details (let me remind you again that our classes are not wrappers around the `libtorrent` classes). Because of this mistake you move part of the logic in the calling code:
```c++
torrent->info().filesForPiece(peer.downloadingPieceIndex())
```
, although it should contain only following:
```c++
peer.downloadingFiles()
```